### ISSUE TITLE
Unexpected @keyup triggering on focused input field when navigated to via keyboard "click"

### ISSUE THREAD

**Issue by iangriggs** [nuxt/nuxt#24434]
### Environment

- Operating System: `Darwin`
- Node Version:     `v18.18.2`
- Nuxt Version:     `3.6.1`
- Nitro Version:    `2.5.2`
- Package Manager:  `npm@9.8.1`
- Builder:          `vite`
- User Config:      `ssr`, `nitro`, `app`, `modules`, `security`, `vite`
- Runtime Modules:  `nuxt-security@1.0.0-rc.2`, `nuxt-vitest@0.10.2`
- Build Modules:    `-`

Although we are pinned at Nuxt v3.6.1 'for reasons' the bug can be observed in StackBlitz repro on Nuxt v3.8.2

### Reproduction

https://stackblitz.com/edit/github-6rrfcj?file=pages%2Findex.vue

Note: Issue relates to keyboard navigation! - not mouse.

- Open `index` page and tab bring focus to button labelled `Goto Two`. 
- Hit enter key to effectively `click` button.  (remember, keyboard - not mouse!)
- Browser navigates to page `two`

Notice the header suggests that the function `handleNext` has been called. This suggest that the `@keyup` event has fired on the `input` control on page `two` as `handleNext` is the function handling of that event. I can't think why this should be expected behaviour. Almost as if the `keyup` event from pressing the button on the `index` page is being seen by page `two` when the `focus()` call is made?! 

Open the browser devtools to see console.log messages also indicating that `handleNext` has been called.

Also, to add additional context/relevant to help diagnose the error. In the script for `two.vue` there is another option (2) that can be uncommented and option (1) commented out. This only calls `focus()` after a delay. With this in place the potential bug doesn't happen.

### Describe the bug

I'm seeing what I think maybe a bug when navigating to a page that combines and `input` control with an associated `@keyup` handler that is also focused on page load via `onMounted` lifecycle event.

Follow the reproduction above and also the effect of delaying the call to `focus()` on the `input` control.

### Additional context

Changing the event type on the `input` field from `@keyup` to `@keydown` also avoids the issue.

**Comment by danielroe** [comment]
I doubt this is a Nuxt issue. Can you replicate with Vue?

https://stackblitz.com/github/nuxt-contrib/vue3-ssr-starter

**Comment by iangriggs** [comment]
Hey @danielroe. Many thanks for responding to this so quickly. I followed your link and re-created in Vue. 

[stackblitz](https://stackblitz.com/edit/github-qhj1am?file=src%2Fpages%2FTwo.vue)

Seems to do the same thing so I guess I'll post the same issue on https://github.com/vuejs/core/issues?

**Comment by warflash** [comment]
Yes, that'd be great! Thank you @iangriggs

**Comment by iangriggs** [comment]
Top help anyone else coming here with a similar concern:

https://github.com/vuejs/core/issues/9696

TL;DR; It's expected behaviour with a couple of workarounds described in comments.
